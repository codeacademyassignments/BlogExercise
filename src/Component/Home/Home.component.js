import React,{Component} from 'react';
import Footer from '../Footer/Footer.component';
import Header from '../Header/Header.component';
import Section from '../Section/Section.component';
import '../index.css';
import axios from 'axios';
// import ReactDOM from 'react-dom';
// import RawData from '../../RawData/RawData';

class Home extends Component{
  state={
    blogs:[],
  };
  createRows = (blogs)=>{
    const rows = [];
    blogs.forEach((blog,index)=>{
      rows.push(<Section key={`${index}${blog.readingTime}`} blog={blog}/>);
    });
    return rows;
  }
  render(){  
    return (
      <section>
      <Header/>
      <section className="main-block">
      {this.createRows(this.state.blogs)}
      </section>
      <section className="main-block">
      <button className='createPost' onClick={()=>window.location.href='CreatePost'}>Create Post</button>
      </section>
      <Footer/>
  </section>
    );
  }

  componentDidMount(){
    const dataPromise = axios.get('https://api.myjson.com/bins/hc5ye');
    // ReactDOM.render(<Header/>,document.getElementById('root'));
    return dataPromise.then((response)=>{
      //console.log(response);
      this.setState({blogs:response.data});
    })
  }
}


export default Home;