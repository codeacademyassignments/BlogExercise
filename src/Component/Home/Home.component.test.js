import React from 'react';
import { shallow } from 'enzyme';
import axios from 'axios';
import renderer from 'react-test-renderer';
import Home from './Home.component';
import testData from '../../RawData/RawData';

let mockAxios;
beforeAll(() => {
  mockAxios = jest.fn().mockImplementation(() => new Promise(((resolve) => { resolve({ data: testData }); })));
  axios.get = mockAxios;
});
afterAll(() => {
  jest.resetModules();
});

describe('Home', () => {
  it('snapshot should match', () => {
    const tree = renderer.create(<Home />);
    expect(tree).toMatchSnapshot();
  });
});

describe('componentDidMount', () => {
  it('should call mock function with proper arguements', async (done) => {
    const wrapper = shallow(<Home />);
    await wrapper.instance().componentDidMount();
    expect(mockAxios).toHaveBeenCalledWith('https://api.myjson.com/bins/hc5ye');
    done();
  });

  it('should update the state of Home component', async (done) => {
    const wrapper = shallow(<Home />);
    expect(wrapper.instance().state.blogs.length).toEqual(0);
    await wrapper.instance().componentDidMount();
    // console.log(wrapper.instance());
    expect(wrapper.instance().state.blogs.length).toEqual(6);
    done();
  });

  it('Snapshot should match', () => {
    const tree = renderer.create(<Home />);
    expect(tree).toMatchSnapshot();
  });
});
