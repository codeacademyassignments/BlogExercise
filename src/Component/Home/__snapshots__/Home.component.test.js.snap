// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapshot should match 1`] = `
<section>
  <header
    className="header"
  >
    <section
      className="main-block"
    >
      <p
        className="main-heading"
      >
        The Artifact
      </p>
      <p
        className="sub-heading"
      >
        Culture & Art Blog
      </p>
      <nav
        className="nav"
      >
        <ul
          className="nav-bar"
        >
          <li
            className="nav-element last-element"
          >
            Blog
          </li>
          <li
            className="nav-element"
          >
            About
          </li>
          <li
            className="nav-element"
          >
            Contact
          </li>
        </ul>
      </nav>
    </section>
  </header>
  <section
    className="main-block"
  />
  <section
    className="main-block"
  >
    <button
      className="createPost"
      onClick={[Function]}
    >
      Create Post
    </button>
  </section>
  <footer
    className="footer"
  >
    <section
      className="main-block footer-section"
    >
      <p
        className="footer-para"
      >
        @artifact.com
         
        <span>
          2019
        </span>
      </p>
    </section>
  </footer>
</section>
`;

exports[`componentDidMount Snapshot should match 1`] = `
<section>
  <header
    className="header"
  >
    <section
      className="main-block"
    >
      <p
        className="main-heading"
      >
        The Artifact
      </p>
      <p
        className="sub-heading"
      >
        Culture & Art Blog
      </p>
      <nav
        className="nav"
      >
        <ul
          className="nav-bar"
        >
          <li
            className="nav-element last-element"
          >
            Blog
          </li>
          <li
            className="nav-element"
          >
            About
          </li>
          <li
            className="nav-element"
          >
            Contact
          </li>
        </ul>
      </nav>
    </section>
  </header>
  <section
    className="main-block"
  />
  <section
    className="main-block"
  >
    <button
      className="createPost"
      onClick={[Function]}
    >
      Create Post
    </button>
  </section>
  <footer
    className="footer"
  >
    <section
      className="main-block footer-section"
    >
      <p
        className="footer-para"
      >
        @artifact.com
         
        <span>
          2019
        </span>
      </p>
    </section>
  </footer>
</section>
`;
